### ISSUE TITLE
Actual leaf area expansion

### ISSUE THREAD

**Issue by byzheng** [APSIMInitiative/ApsimX#2631]
@hol353 and @HamishBrownPFR  

I am trying to figure out why the actual leaf area is not equal to max leaf area in the cohort parameters  under no stress (cell division, expansion, carbon). All stresses have been disabled in a test simulation. However, the leaf size (actual live leaf size) is much bigger the parameter MaxArea  for Leaf 1 (400 vs 346), but smaller for leaf 2 (716 vs 770).  

For leaf 2, I have traced into Structure.cs ([L549-L522](https://github.com/APSIMInitiative/ApsimX/blob/3b4441a0514487c51281c1d02cf20cb3105052d7/Models/Plant/Structure.cs#L549)) which set the cohort Age into 9.13 Cd at appearance. Consequently, the leaf growth duration is reduced 9.13 Cd, then potential leaf size is reduced too. I don't understand why the CohortAge is set as a function of phyllochron at leaf cohort appearance. Could you explain the logic of these codes?

For leaf 1, I guess it is related with PotentialAreaGrowthFunction in [LeafCohort.cs](https://github.com/APSIMInitiative/ApsimX/blob/3b4441a0514487c51281c1d02cf20cb3105052d7/Models/Plant/Organs/LeafCohort.cs#L1015). The `Age + tt` should be capped by `GrowthDuration`, although `SizeFunction` has same slight different to reach MaxArea. I can submit a pull request for this change.

**Comment by byzheng** [comment]
@hol353 @HamishBrownPFR Any comments about this issue?

**Comment by HamishBrownPFR** [comment]
@byzheng is this issue specifically for leaves 1 and 2?  If so they are the ones that are initialised with some area so the problem will be in the initialisation calculations some where

**Comment by byzheng** [comment]
@HamishBrownPFR I think there are two problems.

1) The cohort Age at appearance is initialized as a function of main stem appearance rate "(PotLeafTipsAppeared - TipToAppear) * MainStemNodeAppearanceRate.Value;". Consequently, the actual growth duration is not equal to the parameter GrowthDuration. For Leaf 1, the actual growth duration is equal GrowthDuration. For Leaf 2 and others, the actual growth duration is smaller than GrowthDuration. (Not include in the pull request)

2) In the last day of expansion, the Age + TT is not capped by GrowthDuration. Consequently, in some leaves, the actual growth duration is longer than the parameter GrowthDuration (include in the pull request).

I also attached the example apsimx file [leaf-area-cohort-1.zip](https://github.com/APSIMInitiative/ApsimX/files/2251566/leaf-area-cohort-1.zip) I used in which I have provided enough carbon, water, and nitrogen to make sure there are no any stress to constrain leaf expansion. 

Check P4 and 5 generated by master branch [potential-leaf-size-master.pdf](https://github.com/APSIMInitiative/ApsimX/files/2251565/potential-leaf-size-master.pdf). Leaf 1, 4, and 7 is bigger than potential. Leaf 2, 3, 8 and 9 are smaller than potential.

Check P4 and 5 generated with this pull request plus setting cohort age equal to zero at appearance [potential-leaf-size-new.pdf](https://github.com/APSIMInitiative/ApsimX/files/2251564/potential-leaf-size-new.pdf). All leaves have the same potential and actual leaf area.

Call me if you want to discuss more.
